utils/metamask.ts

```diff
@@ -0,0 +1,262 @@
+    rpc: {
+      metamask: {
+        get_eip712_sign_data: {
+          description: "",
+          params: [
+            {
+              name: "call",
+              type: "String",
+            },
+          ],
+          type: "String",
+        },
+      },
+    },
+    types: {
+      MultiSignature: {
+        _enum: {
+          Ed25519: "Ed25519Signature",
+          Sr25519: "Sr25519Signature",
+          Ecdsa: "EcdsaSignature",
+          Eth: "EcdsaSignature",
+        },
+      },
+      ShufflingSeed: {
+        seed: "H256",
+        proof: "H512",
+      },
+      Header: {
+        parentHash: "Hash",
+        number: "Compact<BlockNumber>",
+        stateRoot: "Hash",
+        extrinsicsRoot: "Hash",
+        digest: "Digest",
+        seed: "ShufflingSeed",
+        count: "BlockNumber",
+      },
+    },
+  });
+  const extrinsic = api.createType(
+    "Extrinsic",
+    { method: tx.method },
+    { version: tx.version },
+  );
+  const dotAddress = blake2AsU8a(hexToU8a(ethAddress));
+  testLog
+    .getLog()
+    .info("dot addr:: " + encodeAddress(blake2AsU8a(hexToU8a(ethAddress)), 42));
+  const options = {};
+  const signingInfo = await api.derive.tx.signingInfo(
+    // @ts-ignore
+    dotAddress,
+    // @ts-ignore
+    options.nonce,
+    // @ts-ignore
+    options.era,
+  );
+  const eraOptions = makeEraOptions(api, api.registry, options, signingInfo);
+  const tx_payload = extrinsic.inner.signature.createPayload(
+    // @ts-ignore
+    extrinsic.method,
+    eraOptions,
+  );
+  const raw_payload = tx_payload.toU8a({ method: true });
+  // @ts-ignore
+  const result = await api.rpc.metamask.get_eip712_sign_data(
+    tx.toHex().slice(2),
+  );
+  console.log(JSON.stringify(result));
+  const data = JSON.parse(result.toString());
+  data.message.tx = u8aToHex(raw_payload).slice(2);
+
+  const msg_sig = eth_sig_utils.signTypedData({
+    privateKey: eth_util.toBuffer(ethPrivateKey),
+    data: data,
+    // @ts-ignore
+    version: "V4",
+  });
+  console.log("Ok, signed typed data ");
+  console.log("SIGNATURE = " + msg_sig);
+  const created_signature = api.createType("MultiSignature", {
+    Eth: hexToU8a(msg_sig),
+  });
+  console.log(tx_payload);
+  console.log(msg_sig);
+  // @ts-ignore
+  extrinsic.addSignature(dotAddress, created_signature, tx_payload);
+  testLog.getLog().info("Sending tx");
+  const txHash = await api.rpc.author.submitExtrinsic(extrinsic.toHex());
+  testLog.getLog().info("Tx sent " + txHash);
+  let extrinsicResult: any;
+  let counter = 10;
+  return new Promise(async (resolve, reject) => {
+    const unsub = await api.rpc.chain.subscribeNewHeads(
+      async (header): Promise<void> => {
+        if (counter <= 0) {
+          unsub();
+          reject("Timeout");
+        }
+        counter--;
+        const currentBlock = await api.rpc.chain.getBlock(header.hash);
+        const extrinsics = currentBlock.block.extrinsics;
+        extrinsicResult = extrinsics.filter(
```

Comment:
add here another 
//TODO: Find some extra determinist way to filter the extrinsic signed by the user.
:pray:

Reply:
Done